File: BOJ/BOJ_1931_회의실배정.py
N = int(input())
conf = [list(map(int, input().split())) for _ in range(N)]
conf.sort(key=lambda x:(x[1], x[0]))
last = conf[0][1]
cnt = 1

for i in range(1, N):
    if conf[i][0] >= last:
        cnt += 1
        last = conf[i][1]

print(cnt)

File: BOJ/BOJ_1931_회의실배정.cpp
#include <bits/stdc++.h>
using namespace std;
typedef pair<int, int> pii;

int N, cnt = 1, last;
vector<pii> conf;

void init() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
}

bool compare(pii &a, pii &b) {
    if (a.second == b.second) {
        return a.first < b.first;
    }
    return a.second < b.second;
}

int main(int argc, char const *argv[]) {
    init();
    cin >> N;
    for (int i = 0; i < N; i++) {
        int s, e; cin >> s >> e;
        conf.push_back({s, e});
    }
    sort(conf.begin(), conf.end(), compare);
    last = conf[0].second;

    for (int i = 1; i < N; i++) {
        if (conf[i].first >= last) {
            cnt++;
            last = conf[i].second;
        }
    }
    cout << cnt << '\n';
    return 0;
}